src/implementation/Server/HTTPServer/HTTPServerImpl.ts

```diff
@@ -40,6 +37,38 @@ export default class HTTPServerImpl {
     this.pubSubSubscriptions = {};
   }
 
+  /**
+   * Extracts data from the subscribe request body.
+   * Data should be present in req.body.data or req.body.data_base64 depending on the rawPayload metadata.
+   * If the data is not found, the request body is returned as-is.
+   * @param req HTTP Request
+   * @returns Data from the request body.
+   */
+  extractSubscribeDataFromRequest(req: any): any {
+    const payload = req.body ?? "";
+
+    // The payload should be an object with string keys and any values.
+    // If data is present in req.body.data, use that as-is.
+    // Else, data is present in req.body.data_base64, and decode it from base64.
+    if (payload instanceof Object && !(payload instanceof Array || payload instanceof Buffer)) {
```

Comment:
Change to a negate, instead of `if (payload instanceof Object && !(payload instanceof Array || payload instanceof Buffer))` let's maybe use `if (!(payload instanceof Object) || payload instanceof Array || payload instanceof Buffer)` this way we can early return removing an extra identation